weston/weston-init

```diff
@@ -28,11 +28,72 @@ startup-animation=fade
 
 IEOF
 
+WPE_WESTON_OUTPUT_MAX_W=${WPE_WESTON_OUTPUT_MAX_W:-0}
+WPE_WESTON_OUTPUT_MAX_H=${WPE_WESTON_OUTPUT_MAX_H:-0}
+
+# Limit resolution if necessary
+if [ "$WPE_WESTON_OUTPUT_MAX_W" -gt 0 -o "$WPE_WESTON_OUTPUT_MAX_H" -gt 0 ]; then
+    for outp in /sys/class/drm/card*-*; do
+        # no outputs
+        if [ ! -d "$outp" ]; then
+            break
+        fi
+        # skip if not enabled
+        if [ "$(cat ${outp}/enabled)" = "disabled" ]; then
+            continue
+        fi
+        # get the output name
+        outn=${outp#*-}
+        # check the modes
+        moden=1
+        foundres=
+        for mode in $(cat "${outp}/modes"); do
+            wres=${mode%x*}
+            hres=${mode#*x}
+            # skip interlaced modes and other junk
+            if ! [ "$wres" -eq "$wres" -a "$hres" -eq "$hres" ] 2> /dev/null; then
```

Comment:
WDYT?:

```sh
re='^[0-9]+$'
for  mode  in $(cat /sys/class/drm/card1-HDMI-A-1/modes)
do
    wres=${mode%x*}
    hres=${mode#*x}
    if ! [[ "$wres" =~ $re  && "$hres" =~ $re ]] 
    then
        # ...
    done
```

Reply:
that's a bashism

Reply:
(mind you, this is a pure /bin/sh script, running in a container with a busybox userland)

Reply:
OK you are right. I wasn't having that in mind:

```sh
is_int() {
    printf %d "$1" >/dev/null 2>&1
}

for  mode in $(cat "${outp}/modes")
do
    wres=${mode%x*}
    hres=${mode#*x}
    !( is_int "$wres" && is_int "$hres" ) && continue

    # ...
done
```

?

Of course, this change isn't a big improve in sake of readability.